lib/Sema/CSBindings.cpp

```diff
@@ -1358,6 +1359,7 @@ void PotentialBindings::infer(Constraint *constraint) {
   case ConstraintKind::BindParam:
   case ConstraintKind::BindToPointerType:
   case ConstraintKind::Subtype:
+  case ConstraintKind::SubclassOf:
```

Comment:
This is actually interesting, I wonder if we do want `subclass` to participate in inference because it could mean inferring a superclass too eagerly.

Reply:
Probably! You'll notice we only generate SubclassOf when the left hand side is a pack for now, because a few tests regressed. Let's figure out how to do it all the time soon, because it will clean up diagnostics too (I saw some hacks to deal with the fact that Superclass requirements become Subtype + ConformsTo AnyObject right now)

Reply:
Hm, I was looking at `matchTypes` for cases where "subtype" is used between class types and it should already handle the case where both types are the same (via DeepEquality restriction) and otherwise it would add `Superclass` restriction that would call `matchSuperclassTypes` internally. Besides handling `PackType`s specifically what are other reasons to add this constraint?